File: 4.py
# 4 - Questão:


class Formulario:
    def __init__(self):
        self.nome = ''
        self.idade = 0
        self.salario = 0
        self.sexo = ''
        self.estado_civil = ''

    def informar_dados(self):
        print('- - - Informe os seguintes dados pessoais: - - -')

        self.obter_nome()
        self.obter_idade()
        self.obter_salario()
        self.obter_sexo()
        self.obter_estado_civil()

        self.mostrar_dados()

    def obter_nome(self):
        while True:
            self.nome = input('Nome: ')
            if len(self.nome) < 3:
                print('O nome deve ter mais que 3 caracteres!')
            else:
                break

    def obter_idade(self):
        while True:
            self.idade = int(input('Idade: '))
            if self.idade < 0 or self.idade > 150:
                print('O idade deve estar entre 0 e 150 anos!')
            else:
                break

    def obter_salario(self):
        while True:
            self.salario = int(input('Salário: '))
            if self.salario < 0:
                print('O salário deve ser maior que zero!')
            else:
                break

    def obter_sexo(self):
        while True:
            self.sexo = input('Sexo: ').lower()
            if self.sexo not in ['f', 'm']:
                print("O sexo deve ser 'f' ou 'm'")
            else:
                break

    def obter_estado_civil(self):
        while True:
            self.estado_civil = input('Estado Civil: ').lower()
            if self.estado_civil not in ['s', 'c', 'v', 'd']:
                print("O estado civil deve ser 's', 'c', 'v' ou 'd'")
            else:
                break

    def mostrar_dados(self):
        print('\n- - - Dados recebidos: - - -')
        print('Nome:', self.nome)
        print('Idade:', self.idade)
        print('Salário:', self.salario)
        print('Sexo:', self.sexo)
        print('Estado Civil:', self.estado_civil)


if __name__ == "__main__":
    main = Formulario()
    main.informar_dados()

File: 4.rb
# 4 - Questão:


class Formulario
    @@nome = ""
    @@idade = 0
    @@salario = 0
    @@sexo = 0
    @@estado_civil = ""

    def informar_dados
        puts "\n- - - Informe os seguintes dados pessoais: - - -"
        
        obter_nome
        obter_idade
        obter_salario
        obter_sexo
        obter_estado_civil

        mostrar_dados
    end

    def obter_nome
        while true
            print "Nome: "
            @@nome = gets.chomp

            if @@nome.length < 3
                puts "O nome deve ter mais que 3 caracteres!"
            else
                break
            end
        end
    end

    def obter_idade
        while true
            print "Idade: "
            @@idade = gets.chomp.to_i

            if @@idade < 0 or @@idade > 150
                puts "A idade deve estar entre 0 e 150 anos!"
            else
                break
            end
        end
    end

    def obter_salario
        while true
            print "Salário: "
            @@salario = gets.chomp.to_i

            if @@salario < 0
                puts "O salário deve ser maior que zero!"
            else
                break
            end
        end
    end

    def obter_sexo
        while true
            print "Sexo: "
            @@sexo = gets.chomp.downcase

            if not ["f", "m"].include?(@@sexo)
                puts "O sexo deve ser 'f' ou 'm'."
            else
                break
            end
        end
    end

    def obter_estado_civil
        while true
            print "Estado Civil: "
            @@estado_civil = gets.chomp.downcase

            if not ["s", "c", "v", "d"].include?(@@estado_civil)
                puts "O estado civil deve ser 's', 'c', 'v' ou 'd'."
            else
                break
            end
        end
    end

    def mostrar_dados
        puts "\n- - - Dados recebidos: - - -"
        puts "Nome: #{@@nome}"
        puts "Idade: #{@@idade}"
        puts "Salário: #{@@salario}"
        puts "Sexo: #{@@sexo}"
        puts "Estado Civil: #{@@estado_civil}"
    end
end

main = Formulario.new
main.informar_dados
